lib/cloud/asset.js

```diff
@@ -47,3 +53,150 @@ export function staticAssetHandler(req) {
     body: data
   });
 }
+
+class Signer {
+  sign() {
+    throw new Error('Not implemented, subclass should override this method');
+  }
+}
+
+/* eslint-disable camelcase */
+class FSSigner extends Signer {
+  constructor(_settings) {
+    super();
+    this.settings = _settings;
+  }
+
+  sign(name) {
+    const prefix = this.settings.assetStoreURLPrefix;
+    const duration = this.settings.assetStoreURLExpireDuration;
+    const expire = Math.floor(Date.now() / 1000) + duration;
+    const secret = this.settings.assetStoreSecret;
```

Comment:
If we assume settings won't change. Just copy the setting into the `this`, not querying everytime during sign?